File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans =[]
        self.traverse(root,ans)
        return ans
    
    def traverse(self,node,ans):
        if node is None:
            return
        
        self.traverse(node.left,ans)
        self.traverse(node.right,ans)
        ans.append(node.val)

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    int max =0;
    public int diameterOfBinaryTree(TreeNode root) {
        findHeight(root);
        return max;
    }

    private int findHeight(TreeNode root){
        if (root == null){
            return 0;
        }

        int left = findHeight(root.left);
        int right = findHeight(root.right);

        max = Math.max(max, left + right);

        return 1 + Math.max(left,right);
    }
}
